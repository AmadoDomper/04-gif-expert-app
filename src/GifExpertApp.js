import React, { useState } from 'react'
import { AddCategory } from './Components/AddCategory';


const GifExpertApp = () => {

    // const categories = ['One punch','Samurai X','Dragon Ball','DB Z Kaie'];
    const [categories, setCategories] = useState(['One punch','Samurai X','Dragon Ball','DB Z Kaie']);

    // const HandleAdd = () => {
    //     // setCategories([...categories, 'HunterXHunter']);
    //     setCategories(cats => [...cats, 'HunterXHunter']);
    //     console.log(categories);
    // }

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={setCategories} />
            <hr />

            <ol>
                {
                    categories.map(category => {
                        return <li key={category}>{category}</li>
                    })
                }
            </ol>
        </>
    )
}

export default GifExpertApp;