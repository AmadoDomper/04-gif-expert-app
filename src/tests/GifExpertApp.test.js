
import { shallow } from "enzyme";
import '@testing-library/jest-dom';
import GifExpertApp from "../GifExpertApp";

describe('Test for <GifExpertApp />', () => {
    
    test('should show <GifExpertApp /> correctly', () => {
        const wrapper = shallow(<GifExpertApp />);

        expect(wrapper).toMatchSnapshot();
    })

    test('should show a list of category', () => {
        const categories = ['One Punch','Dragon Ball'];

        const wrapper = shallow(<GifExpertApp defaultCategories={ categories } />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("GifGrid").length).toBe( categories.length );
    })
    
})
