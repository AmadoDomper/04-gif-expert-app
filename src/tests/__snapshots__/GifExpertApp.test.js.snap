// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for <GifExpertApp /> should show <GifExpertApp /> correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Test for <GifExpertApp /> should show a list of category 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
